crates/holochain/tests/tests/zero_arc/mod.rs

```diff
@@ -384,3 +390,103 @@ async fn self_validation_get_missing() {
         "must_get_entry was not invoked"
     );
 }
+
+/// Tests that after deleting a link, later calls to get_links will not return the deleted link.
+#[tokio::test(flavor = "multi_thread")]
+async fn zero_arc_delete_link_get_links() {
+    test_run();
+
+    let entry_def = EntryDef::default_from_id("entry");
+    let zomes = SweetInlineZomes::new(vec![entry_def], 1)
+        .function("create_link", move |api, base: ActionHash| {
+            let hash = api.create_link(CreateLinkInput::new(
+                base.into(),
+                fixt::fixt!(ActionHash).into(),
+                0.into(),
+                0.into(),
+                vec![].into(),
+                ChainTopOrdering::default(),
+            ))?;
+            Ok(hash)
+        })
+        .function("delete_link", move |api, link_hash: ActionHash| {
+            let _ = api.delete_link(DeleteLinkInput::new(
+                link_hash,
+                GetOptions::default(),
+                ChainTopOrdering::default(),
+            ))?;
+            Ok(())
+        })
+        .function("get_links", move |api, base: ActionHash| {
+            let links = api.get_links(vec![GetLinksInput::from_query(
+                LinkQuery::new(base, LinkTypeFilter::single_type(0.into(), 0.into())),
+                GetOptions::default(),
+            )])?;
+            Ok(links.into_iter().flatten().collect::<Vec<_>>())
+        });
+
+    let base = fixt::fixt!(ActionHash);
+
+    let (test_dna, _, _) = SweetDnaFile::unique_from_inline_zomes(zomes).await;
+    let standard_conductor_config = SweetConductorConfig::rendezvous(false);
+    let zero_arc_conductor_config =
+        SweetConductorConfig::rendezvous(false).tune_network_config(|nc| {
+            nc.target_arc_factor = 0;
+        });
+
+    let mut conductors = SweetConductorBatch::from_configs_rendezvous([
+        standard_conductor_config,
+        zero_arc_conductor_config,
+    ])
+    .await;
+
+    let apps = conductors.setup_app("test_app", [&test_dna]).await.unwrap();
+
+    let ((alice,), (bob,)) = apps.into_tuples();
+
+    let alice_zome = alice.zome(SweetInlineZomes::COORDINATOR);
+    let bob_zome = bob.zome(SweetInlineZomes::COORDINATOR);
+
+    // Alice creates a link
+    let link_hash: ActionHash = conductors[0]
+        .call(&alice_zome, "create_link", base.clone())
+        .await;
+
+    // Simulate Alice reaching a full storage arc
+    conductors[0]
+        .declare_full_storage_arcs(test_dna.dna_hash())
+        .await;
+
+    // and ensure Bob knows about Alice's full arc.
+    conductors.exchange_peer_info().await;
+
+    // Wait for Bob to see the link
+    retry_fn_until_timeout(
+        || async {
+            let links: Vec<Link> = conductors[1]
+                .call(&bob_zome, "get_links", base.clone())
+                .await;
+            if links.iter().any(|link| link.create_link_hash == link_hash) {
+                return true;
+            }
+
+            false
```

Comment:
```suggestion
            links.iter().any(|link| link.create_link_hash == link_hash)
```

Reply:
Don't know why I have such trouble seeing this!

https://github.com/holochain/holochain/pull/5421/commits/9a11151d3457560d1a728a7a02235d4a5792c5bf